core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala

```diff
@@ -173,11 +173,11 @@ abstract class KafkaServerTestHarness extends QuorumTestHarness {
     adminClientConfig: Properties = new Properties
   ): scala.collection.immutable.Map[Int, Int] = {
     if (isKRaftTest()) {
-      resource(createAdminClient(brokers, listenerName, adminClientConfig)) { admin =>
+      resource(createAdminClient(aliveBrokers, listenerName, adminClientConfig)) { admin =>
```

Comment:
Is this change necessary? I think AdminClient should be robust enough to recognize dead and alive brokers and handle it properly.

Reply:
I agree. Let's revert the changes in this file.

Reply:
The problem is we need at least one change in `deleteTopic()`. Because it checks to see if metadata is propagated to all the brokers in `(brokers, topic, 0)`. This will fail if one of the `brokers` passed to `waitForAllPartitionsMetadata` is dead.

Reply:
Let me know your thoughts. I removed the check while giving it to create adminClient but still kept in the places we check for metadata propagation.

Reply:
If that is the case, then you should wait until all brokers are unfenced before proceeding in the test

Reply:
I looked at this again, and I think using aliveBrokers is correct.